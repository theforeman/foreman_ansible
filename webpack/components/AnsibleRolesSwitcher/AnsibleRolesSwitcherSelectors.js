import { differenceBy, slice } from 'lodash';

export const calculateUnassignedRoles = state => differenceBy(state.results, state.assignedRoles, 'id');

export const assignedRolesPage = (assignedRoles, assignedPagination) => {
  const offset = (assignedPagination.page - 1) * assignedPagination.perPage;

  return slice(assignedRoles, offset, offset + assignedPagination.perPage);
};
