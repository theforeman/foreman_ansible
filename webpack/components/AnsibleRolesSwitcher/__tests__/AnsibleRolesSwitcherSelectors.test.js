import { calculateUnassignedRoles, assignedRolesPage } from '../AnsibleRolesSwitcherSelectors';

import { ansibleRolesShort, AnsibleRolesLong } from '../__fixtures__/ansibleRolesData.fixtures';

describe('calculateUnassignedRoles', () => {
  it('should return unassigned roles', () => {
    const assignedRoles = [
      { id: 2 },
      { id: 4 },
    ];

    expect(() => {
      calculateUnassignedRoles({ ansibleRolesShort, assignedRoles }).toEqual([
        { id: 1, name: 'sthirugn.motd' },
        { id: 3, name: 'rvm.ruby' },
      ]);
    });
  });

  it('should return all roles when no roles assigned', () => {
    expect(() => {
      calculateUnassignedRoles({ results: ansibleRolesShort, assignedRoles: [] })
        .toEqual(ansibleRolesShort);
    });
  });
});

describe('assignedRolesPage', () => {
  it('should return requested page', () => {
    expect(() => {
      assignedRolesPage(AnsibleRolesLong, { page: 2, perPage: 5 })
        .toEqual(AnsibleRolesLong.slice(5, 10));
    });
  });
});
