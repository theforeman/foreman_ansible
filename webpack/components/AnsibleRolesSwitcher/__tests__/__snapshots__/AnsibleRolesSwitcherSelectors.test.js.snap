// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleRolesSwitcherSelectors should return all roles when no roles assigned 1`] = `
Array [
  Object {
    "id": 4,
    "name": "geerlingguy.java",
  },
  Object {
    "id": 2,
    "name": "jtyr.ntp",
  },
  Object {
    "id": 3,
    "name": "rvm.ruby",
  },
  Object {
    "id": 1,
    "name": "sthirugn.motd",
  },
]
`;

exports[`AnsibleRolesSwitcherSelectors should return requested page 1`] = `
Array [
  Object {
    "id": 5,
    "name": "naftulikay.golang",
  },
  Object {
    "id": 8,
    "name": "puppet.puppet",
  },
  Object {
    "id": 3,
    "name": "rvm.ruby",
  },
  Object {
    "id": 10,
    "name": "salt.salt",
  },
  Object {
    "id": 1,
    "name": "sthirugn.motd",
  },
]
`;

exports[`AnsibleRolesSwitcherSelectors should return unassigned roles 1`] = `
Array [
  Object {
    "id": 3,
    "name": "rvm.ruby",
  },
  Object {
    "id": 1,
    "name": "sthirugn.motd",
  },
]
`;
